test/integration/inbound/dfsp-linking.test.ts

```diff
@@ -0,0 +1,1319 @@
+                  "aute irure dolor in reprehenderit in voluptate velit " +
+                  "esse cillum dolore eu fugiat nulla pariatur."
+              },
+              "type": "public-key"
+            }
+          }
+        }
+
+        const putScenarioUri = `${env.inbound.baseUri}/consents/${consentId}`
+        const responseToPutConsents = await axios.put(putScenarioUri, putConsentsIDSignedCredentialPayload, axiosConfig)
+        expect(responseToPutConsents.status).toEqual(202)
+
+        await new Promise(resolve => setTimeout(resolve, 200));
+
+        // check that the DFSP has sent a POST /consents to the auth-service
+        const requestsHistory: MLTestingToolkitRequest[] = (await axios.get(ttkRequestsHistoryUri, axiosConfig)).data
+        const postConsentsToAuthService = requestsHistory.filter(req => {
+          return req.method === 'post' && req.path === '/consents'
+        })
+        expect(postConsentsToAuthService.length).toEqual(1)
+        const historyPayload = postConsentsToAuthService[0].body as tpAPI.Schemas.ConsentsPostRequestAUTH
+
+        expect(historyPayload).toEqual(
+          expect.objectContaining({
+            consentId: consentId,
+            credential: expect.any(Object),
+            scopes: expect.any(Object)
+          })
+        )
+      })
+    })
+
+    describe('Inbound PUT /consents/{ID}/error from auth-service', (): void => {
+      it('should send back PUT /consents/{ID}/error to PISP', async (): Promise<void> => {
+        // the auth-service now sends back a PUT /consents/{ID} confirming verification
+        const putConsentsIDErrorPayload = {
+          errorInformation: {
+            errorCode: "7213",
+            errorDescription: "Consent is invalid"
+          }
+        }
+
+        const putScenarioUri = `${env.inbound.baseUri}/consents/${consentId}/error`
+        const responseToPutConsents = await axios.put(putScenarioUri, putConsentsIDErrorPayload, axiosConfig)
+        expect(responseToPutConsents.status).toEqual(200)
+
+        await new Promise(resolve => setTimeout(resolve, 200));
+
+        // check that the DFSP has sent a PUT /consents/{ID}/error to the PISP
+        const requestsHistory: MLTestingToolkitRequest[] = (await axios.get(ttkRequestsHistoryUri, axiosConfig)).data
+        const putConsentsErrorToPISP = requestsHistory.filter(req => {
+          return req.method === 'put' && req.path === `/consents/${consentId}/error`
+        })
+        expect(putConsentsErrorToPISP.length).toEqual(1)
+        const historyPayload = putConsentsErrorToPISP[0].body as tpAPI.Schemas.ErrorInformation
+
+        expect(historyPayload).toEqual(
+          {
+            errorInformation: {
+              errorCode: "7213",
+              errorDescription: "Consent is invalid"
+            }
+          }
+        )
+      })
+    })
+  })
+
+  describe('DFSP recieves error from ALS after sending the auth-service PISP PENDING credential', (): void => {
```

Comment:
I'm not sure I followed this error case. What is supposed to happen after `Inbound PUT /consents/{ID} signed credential from PISP`? Are we meant to be testing when the auth-service fails to register the credential with the ALS?

Reply:
deleted.